tests/Config/ConfigurationLoaderTest.php

```diff
@@ -125,205 +127,149 @@ function (?string $eTag) use ($flagsResourceResponse, $flagsRaw) {
-        $loader = new ConfigurationLoader($apiWrapper, $configStore, optimizedBanditLoading: true);
-
-
-        // First fetch has the bandit cold
-        $loader->fetchAndStoreConfigurations(null);
-
-        $bandit = $loader->getBandit('cold_starting_bandit');
-        $this->assertNotNull($bandit);
-        $this->assertInstanceOf(Bandit::class, $bandit);
-        $this->assertEquals('cold_starting_bandit', $bandit->banditKey);
-        $this->assertEquals('cold start', $bandit->modelVersion);
-
-
-        // Trigger a reload, second fetch shows the bandit as still cold
-        $loader->fetchAndStoreConfigurations('initial');
-
-        $bandit = $loader->getBandit('cold_starting_bandit');
-        $this->assertNotNull($bandit);
-        $this->assertInstanceOf(Bandit::class, $bandit);
-        $this->assertEquals('cold_starting_bandit', $bandit->banditKey);
-        $this->assertEquals('cold start', $bandit->modelVersion);
-
-        // Trigger a reload, third fetch has the bandit warm with v1
-        $loader->fetchAndStoreConfigurations('initialButForced');
-
-        $bandit = $loader->getBandit('cold_starting_bandit');
-        $this->assertNotNull($bandit);
-        $this->assertInstanceOf(Bandit::class, $bandit);
-        $this->assertEquals('cold_starting_bandit', $bandit->banditKey);
-        $this->assertEquals('v1', $bandit->modelVersion);
+        $this->assertEquals($timestamp1, $configStore->getConfiguration()->getFetchedAt());
     }
 
-    public function testReloadsOnExpiredCache(): void
-    {
-        // Arrange: Load some flag data to be returned by the APIRequestWrapper
-        // Load mock response data
-        $flagsRaw = file_get_contents(self::MOCK_RESPONSE_FILENAME);
-        $flagsJson = json_decode($flagsRaw, true);
-        $banditsRaw = '{"bandits": {}}';
-
-        $apiWrapper = $this->getMockBuilder(APIRequestWrapper::class)->disableOriginalConstructor()->getMock();
-
-        $cache = DefaultCacheFactory::create();
-        // Act: Create a new FCL with a 0sec ttl and retrieve a flag
-        $loader = new ConfigurationLoader($apiWrapper, new ConfigurationStore($cache), cacheAgeLimitMillis: 0);
-
-        // Mocks verify interaction of loader <--> API requests and loader <--> config store
-        $apiWrapper->expects($this->exactly(2))
-            ->method('getUFC')
-            ->willReturn(new APIResource($flagsRaw, true, "ETAG"));
-        $apiWrapper->expects($this->exactly(2))
-            ->method('getBandits')
-            ->willReturn(new APIResource($banditsRaw, true, "ETAG"));
-
-        $flag = $loader->getFlag(self::FLAG_KEY);
-        $flagAgain = $loader->getFlag(self::FLAG_KEY);
-
-        // Assert: non-null flag, api called only once via Mock `expects` above.
-        $this->assertNotNull($flag);
-        $this->assertNotNull($flagAgain);
-        $this->assertEquals($flag, $flagAgain);
-    }
+//    public function testOnlyLoadsBanditsWhereNeeded(): void
```

Comment:
👀 
I bet suporting this is relatively easy--essentially a set inclusion check. Consider a fast-fallow PR while this is fresh.

Reply:
It is, I just figured it was a nice and easy piece to separate out in another commit/PR